src/pages/storage/StorageVolumeSnapshots.tsx

```diff
@@ -0,0 +1,340 @@
+        onSuccess={onSuccess}
+        onFailure={onFailure}
+      />
+    );
+
+    return {
+      className: "u-row",
+      name: snapshot.name,
+      columns: [
+        {
+          content: (
+            <>
+              <div className="u-truncate" title={snapshot.name}>
+                <ItemName item={snapshot} />
+              </div>
+              {isSmallScreen && (
+                <div className="u-text--muted">
+                  {isoTimeToString(snapshot.created_at)}
+                </div>
+              )}
+            </>
+          ),
+          role: "rowheader",
+          "aria-label": "Name",
+          className: "name",
+        },
+        ...(isSmallScreen
+          ? []
+          : [
+              {
+                content: isoTimeToString(snapshot.created_at),
+                role: "rowheader",
+                "aria-label": "Created at",
+                className: "created",
+              },
+            ]),
+        {
+          content: isoTimeToString(snapshot.expires_at),
+          role: "rowheader",
+          "aria-label": "Expires at",
+          className: "expiration",
+        },
+        {
+          content: actions,
+          role: "rowheader",
+          "aria-label": "Actions",
+          className: "u-align--right actions",
+        },
+      ],
+      sortData: {
+        name: snapshot.name.toLowerCase(),
+        created_at: snapshot.created_at,
+        expires_at: snapshot.expires_at,
+      },
+    };
+  });
+
+  const pagination = usePagination(rows, "created_at", "descending");
+
+  const resize = () => {
+    setSmallScreen(figureCollapsedScreen());
+  };
+  useEventListener("resize", resize);
+
+  if (error) {
+    notify.failure("Loading storage volume snapshots failed", error);
+  }
+
+  if (isSnapshotsLoading || isProjectLoading) {
+    return <Loader text="Loading storage volume snapshots..." />;
+  } else if (!volumeSnapshots) {
+    return <>Loading storage volume snapshots failed</>;
+  }
+
+  return (
+    <div className="snapshot-list">
+      {hasSnapshots && (
+        <div className="upper-controls-bar">
+          {selectedNames.length === 0 ? (
+            <>
+              <div className="search-box-wrapper">
+                <SearchBox
+                  name="search-snapshot"
+                  className="search-box margin-right"
+                  type="text"
+                  onChange={(value) => {
+                    setQuery(value);
+                  }}
+                  placeholder="Search for snapshots"
+                  value={query}
+                  aria-label="Search for snapshots"
+                />
+              </div>
+              <VolumeConfigureSnapshotBtn
+                volume={volume}
+                onSuccess={onSuccess}
+                onFailure={onFailure}
+                className="u-no-margin--right"
+              />
+              <VolumeAddSnapshotBtn volume={volume} className="u-float-right" />
```

Comment:
Those two buttons also need to be disabled when the project is restricting snapshot creation. Best with an explainer like on the instance snapshots (see other comment)